src/algebra/lie_algebra.lean

```diff
@@ -195,10 +195,14 @@ variables [lie_algebra R L₁] [lie_algebra R L₂] [lie_algebra R L₃]
 
 instance : has_coe (L₁ →ₗ⁅R⁆ L₂) (L₁ →ₗ[R] L₂) := ⟨morphism.to_linear_map⟩
 
-lemma map_lie (f : L₁ →ₗ⁅R⁆ L₂) (x y : L₁) : f ⁅x, y⁆ = ⁅f x, f y⁆ := morphism.map_lie f
+/-- It may seem redundant that we define a coercion to a function given that we already have
+a coercion to a linear map, which in turn coerces to a function. In fact it is not redundant,
+since otherwise an expression like `f z` is not in simp-normal form, and so should not appear on
+the LHS of a simp lemma (as it does in `map_lie` below). The reason is because Lean collapses
+chains of coercions to a single coercion whereas the simplifier undoes this. -/
```

Comment:
```suggestion
/-- see Note [function coercion] -/
```

We typically try to document these design decisions that occur multiple times scatter over mathlib in so called "library notes".  Feel free to add to / improve upon the corresponding library note for function coercions:
https://github.com/leanprover-community/mathlib/blob/c3d943e2e081c39226d38b62e5b00aeacb84d59e/src/logic/basic.lean#L69-L94

Reply:
The library notes are also published in our beautiful documentation: https://leanprover-community.github.io/mathlib_docs/notes.html#function%20coercion

Reply:
Thanks @gebner definitely prefer this.